server/main.go

```diff
@@ -5,21 +5,53 @@ import (
 	"log"
 	"net"
 
-	pb "github.com/pl0q1n/goDHT/client_proto"
+	"golang.org/x/net/context"
+
+	pbClient "github.com/pl0q1n/goDHT/client_proto"
+	pbNode "github.com/pl0q1n/goDHT/node_proto"
 	api "github.com/pl0q1n/goDHT/server/api"
 
 	"google.golang.org/grpc"
 )
 
 func main() {
+	mode := flag.String("mode", "new", "mode")
```

Comment:
Looks like presence/absence of `target` argument is enough to determine startup type so we don't need this `mode` flag.